src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstallerServices.cs

```diff
@@ -101,6 +104,18 @@ public IEnumerable<IVsPackageMetadata> GetInstalledPackages()
                 });
         }
 
+        private async Task<FallbackPackagePathResolver> GetPackagesPathResolverAsync(BuildIntegratedNuGetProject project)
+        {
+            // To get packagesPath for build integrated projects, first read the packageSpec to know if
+            // RestorePackagesPath property was specified. If yes, then use that property to get packages path
+            // otherwise use global user cache folder from _settings.
+            var context = new DependencyGraphCacheContext();
+            var packageSpec = (await project.GetPackageSpecsAsync(context)).First();
```

Comment:
this should be fine, but you should take the first spec that is of type PackageReference and not a tool spec. 
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.ProjectModel/ProjectStyle.cs#L21